storage-node/src/services/sync/tasks.ts

```diff
@@ -104,15 +109,18 @@ export class DownloadFileTask implements SyncTask {
         await this.tryDownload(chosenBaseUrl, filepath)
 
         // if download succeeds, break the loop
-        if (fs.existsSync(filepath)) {
+        try {
+          await fsPromises.access(filepath, fs.constants.F_OK)
           return
+        } catch (err) {
+          continue
         }
       } catch (err) {
         logger.error(`Sync - fetching data error for ${this.dataObjectId}: ${err}`, { err })
       }
     }
 
-    logger.warn(`Sync - cannot get operator URLs for ${this.dataObjectId}`)
+    logger.warn(`Sync - Failed to download ${this.dataObjectId}`)
```

Comment:
Not part of this PR but shouldn't the severity be the other way around actually? Warning for a single download and error if it wasn't possible to get an object at all?

Reply:
Yeah I think we need to have a PR just for improving the logging.
Also it is a bit more nuanced. We would need better error type checks to know if a download failing is an error, due to a problem from the local operator side, or from the remote node.